src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -262,7 +262,8 @@ void CARMACloudPlugin::Broadcast_TCMs()
 			std::lock_guard<mutex> lock(_not_ACK_TCMs_mutex);
 			for( auto itr = _not_ACK_TCMs->begin(); itr!=_not_ACK_TCMs->end(); ++itr )
 			{
-				string tcmv01_req_id_hex = itr->first;
+				string tcmv01_req_id_hex = itr->first;	
+				//Start recording time out period
 				if(!is_started_broadcasting)

```

Comment:
I don't understand the start_time logic, shouldn't start_time be different for each message because that should be set to the first time the message was added to the queue or broadcast, is_started_broadcasting is set to true after the first message is processed and will stay true for the rest of the collection.  So, all messages will have the same start time and different curr times.